lumicks/pylake/fdcurve.py

```diff
@@ -194,6 +216,16 @@ def with_channels(self, force, distance):
         new_fd._primary_distance_channel = distance
         return new_fd
 
+    def with_baseline_correction(self):
+        """Return a copy of this FD curve with baseline correction applied to force channel.
+        Note, only the x component is available."""
+        new_fd = copy(self)
+        new_fd._baseline_corrected = True
+        new_fd._force_cache = None
+        if len(new_fd.f) == 0:
+            raise ValueError("baseline correction not found.")
+        return new_fd
```

Comment:
now, rather than resetting the caches, the corrected force and distance is read directly from the h5 file and evaluated immediately (similar to what is done in `FdCurve.with_offset()`). I've added an extra note to the docstring and a more explicit description in the tutorial docs.